### DISCUSSION TITLE
Questions about Dependencies and Licenses

### DISCUSSION THREAD

**Discussion by sobrito** [resiprocate/resiprocate#337] [General]
In the process of deepening my understanding of resiprocate and aiming to migrate it to Conan 2, I've encountered some issues that surpass my current level of expertise. I'm seeking your guidance to clarify a few points I've come across along the way:

1. File Dependency and Class Extension:

While analyzing the code, I noticed that the file "resip/recon/SipXMediaStackAdapter.hxx" includes <os/OsMsgDispatcher.h>, a file present in the sipXtapi project, but not in our current files. I also observed that "SipXMediaInterface" seems to extend "OsMsgDispatcher," which I believe to be part of sipXtapi. I’m somewhat confused, as I didn't identify sipXtapi as a dependency of this project. Could you clarify whether this is intentional or if I’m interpreting something incorrectly?

2. Voice Traffic Management:

At the moment, I'm seeking to better understand the functionalities related to voice traffic and wondering if we are tied to using sipXtapi for these operations. Is there a possibility of achieving the desired results without relying on this specific library? If so, could you guide me on how to proceed?

3. Licensing Issues:
I identified that sipXtapi is under the LGPL license, while resiprocate is governed by the Vovida Software License, Version 1.0. If the project indeed requires sipXtapi due to the activation of reTurn or recon, which license should I specify in the Conan recipe?

I thank you in advance for your attention and the time spent helping clarify these doubts.

**Comment by sgodin** [comment]
...inline...

On Thu, Oct 19, 2023 at 4:09 PM Sileno De Oliveira Brito <
***@***.***> wrote:

> In the process of deepening my understanding of resiprocate and aiming to
> migrate it to Conan 2, I've encountered some issues that surpass my current
> level of expertise. I'm seeking your guidance to clarify a few points I've
> come across along the way:
>
>    1. File Dependency and Class Extension:
>
> While analyzing the code, I noticed that the file
> "resip/recon/SipXMediaStackAdapter.hxx" includes <os/OsMsgDispatcher.h>, a
> file present in the sipXtapi project, but not in our current files. I also
> observed that "SipXMediaInterface" seems to extend "OsMsgDispatcher," which
> I believe to be part of sipXtapi. I’m somewhat confused, as I didn't
> identify sipXtapi as a dependency of this project. Could you clarify
> whether this is intentional or if I’m interpreting something incorrectly?
>
Yes, those are sipXtapi files.  The only part of resiprocate that uses
sipXtapi is the recon API layer.  Recon brings together the core
resiprocate stack and a media engine (sipXtapi).  See here for more info:
https://github.com/resiprocate/resiprocate/wiki/recon-Overview

Note:  Relatively recently, Daniel Pocock has been adding more media layer
options to recon, including GStreamer, LibWebRtc and Kurento.  As far as I
know, these are still a work in progress.  The sipXtapi solution is fully
functional.


>    1. Voice Traffic Management:
>
> At the moment, I'm seeking to better understand the functionalities
> related to voice traffic and wondering if we are tied to using sipXtapi for
> these operations. Is there a possibility of achieving the desired results
> without relying on this specific library? If so, could you guide me on how
> to proceed?
>
I suppose that depends on what your desired results are.  :)  If you are
using the core resip stack, repro, return, or reflow, then you do NOT
require sipXtapi.  The SIP stack is entirely decoupled from media
handling.  It is only the recon subproject that contains media operations.


>
>    1. Licensing Issues:
>    I identified that sipXtapi is under the LGPL license, while
>    resiprocate is governed by the Vovida Software License, Version 1.0. If the
>    project indeed requires sipXtapi due to the activation of reTurn or recon,
>    which license should I specify in the Conan recipe?
>
> If you don't have a use for recon, I would suggest you don't bother
including recon in the conan packages.  Here's a list of resip licenses -
it depends on which components you use as to what licenses you need to be
aware of:
https://github.com/resiprocate/resiprocate/wiki/License

recon specific licensing info is here:
https://github.com/resiprocate/resiprocate/wiki/recon-Overview#user-content-Licensing

Best Regards,
Scott Godin
SIP Spectrum, Inc.
http://www.sipspecturm.com
